src/util.h

```diff
@@ -309,15 +310,12 @@ std::string HelpMessageOpt(const std::string& option, const std::string& message
  */
 int GetNumCores();
 
-void RenameThread(const char* name);
-
 /**
  * .. and a wrapper that just calls func once
  */
 template <typename Callable> void TraceThread(const char* name,  Callable func)
```

Comment:
Only problem is that'd create a circular dep with `logging` and `threadutil` due to use of `LogPrintf` here. I agree that in concept that's where it should live, but it doesn't look trivial to avoid that circular dep.